G/GDAL/build_tarballs.jl

```diff
@@ -122,30 +125,33 @@ products = [
 
 hdf5_platforms = [
     Platform("x86_64", "linux"),
-    Platform("aarch64", "linux"; libc="glibc"),
+    Platform("aarch64", "linux"),
+    Platform("armv6l", "linux"),
+    Platform("armv7l", "linux"),
+    Platform("i686", "linux"),
+    Platform("powerpc64le", "linux"),
     Platform("x86_64", "macos"),
+    Platform("aarch64", "macos"),
     Platform("x86_64", "windows"),
     Platform("i686", "windows"),
-    Platform("aarch64", "macos"),
 ]
 hdf5_platforms = expand_cxxstring_abis(hdf5_platforms)
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency("GEOS_jll"; compat="~3.11"),
-    Dependency("PROJ_jll"; compat="~900.100"),
+    Dependency("GEOS_jll"; compat="3.11.2"),
+    Dependency("PROJ_jll"; compat="901.300.0"),
```

Comment:
PROJ_jll above had a `~` compat specification.  @visr what's the usual thing to do here with this package?  Is `^` (the default) enough?

Reply:
Yeah I think this is fine. Since PROJ_jll has the 100x version_offset, I figured in #7366 it's easier to tag it as breaking using that, rather than forcing all dependencies to use `~`.